cores/psoc6/time.c : Refactored delay() and delayMicroseconds() implementation.

jaenrig-ifx opened the pull request:
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Reverted delay functions to original implementation. 
We used systick originally, then it was not working with wifi so we replaced it by vTaskDelay(). 
Then we refactored it based on the cyhal_timer, but keep the vTaskDelay() implementation. And that is blocking the interrupt triggering (we had this behavior in the Wire slave implementation). 
Now with the cyhal_timer implementation, the original millis()/micro() time difference works without any identified flaw.

Manually evaluated the following tests:
- test_digitalio_single
- test_interrupts_single
- test_uart_tx - test_uart_rx
- wifi (all)
- Wire examples (with the previously conflicting delay() introduced again)

ramya-subramanyam commented:
> By creating this pull request you agree to the terms in CONTRIBUTING.md. https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md --- DO NOT DELETE ANYTHING ABOVE THIS LINE ---
> 
> CONTRIBUTING.md also tells you what to expect in the PR process.
> 
> Reverted delay functions to original implementation. We used systick originally, then it was not working with wifi so we replaced it by vTaskDelay(). Then we refactored it based on the cyhal_timer, but keep the vTaskDelay() implementation. And that is blocking the interrupt triggering (we had this behavior in the Wire slave implementation). Now with the cyhal_timer implementation, the original millis()/micro() time difference works without any identified flaw.
> 
> Manually evaluated the following tests:
> 
> * test_digitalio_single
> * test_interrupts_single
> * test_uart_tx - test_uart_rx
> * wifi (all)
> * Wire examples (with the previously conflicting delay() introduced again)

So now the wifi tests also are working?

jaenrig-ifx commented:
> > By creating this pull request you agree to the terms in CONTRIBUTING.md. https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md --- DO NOT DELETE ANYTHING ABOVE THIS LINE ---
> > CONTRIBUTING.md also tells you what to expect in the PR process.
> > Reverted delay functions to original implementation. We used systick originally, then it was not working with wifi so we replaced it by vTaskDelay(). Then we refactored it based on the cyhal_timer, but keep the vTaskDelay() implementation. And that is blocking the interrupt triggering (we had this behavior in the Wire slave implementation). Now with the cyhal_timer implementation, the original millis()/micro() time difference works without any identified flaw.
> > Manually evaluated the following tests:
> > 
> > * test_digitalio_single
> > * test_interrupts_single
> > * test_uart_tx - test_uart_rx
> > * wifi (all)
> > * Wire examples (with the previously conflicting delay() introduced again)
> 
> So now the wifi tests also are working?

Yep. I think the mix of systick based timer was not integrated in the RTOS, but the cyhal_timer is.
